ethereum/PFP.sol

```diff
@@ -147,14 +147,15 @@ contract PFP is ERC721A, Ownable, VRFConsumerBase {
             if (MerkleProof.verify(merkleProof, snapshotMerkleRoot, keccak256(abi.encodePacked(msg.sender, '_', allowance)))) {
                 if(_claimedAmount[coldWallet] + numberOfTokens <= allowance ) {
                     _safeMint(msg.sender, numberOfTokens);
```

Comment:
Using ERC721A minting is cheap, BUT large mint batches can result in very expensive transfers, as the contract needs to determine the owner by iterating back through the ownership index. If you have users minting 100s of tokens they will see high gas costs on the first transfer event after mint.

See this useful article, in particular the guidance in "Tip 1": https://medium.com/@dumbnamenumbers/your-guide-to-erc721a-july-2022-f81f0be84a54. You may want to set an 'internal' batch size (in the article example it uses 10 tokens). This limits the number of iterations required to find the owner on the first transfer call to a maximum of 10. BUT, it does that by writing to storage in the mint (so your mint gets more expensive than if it didn't do this).

This is a trade-off - gas cost at mint vs gas cost in the first transfer after mint. And therefore a judgement call you have to make :).